wfsim/core/s2.py

```diff
@@ -332,42 +330,96 @@ def photon_timings(t, n_electron, z, xy, sc_gain, config, resource, phase):
-                                                             config=config,
-                                                             resource=resource)
-        elif config['s2_luminescence_model'] == 'garfield':
-            _photon_timings = S2.luminescence_timings_garfield(
-                xy, n_photons_per_xy,
-                config=config,
-                resource=resource)
-
-        _photon_timings += np.repeat(_electron_timings, n_photons_per_ele)
         _instruction = np.repeat(np.arange(len(xy)), n_photons_per_xy)
+        # Get photon channels
+        _photon_channels = S2.photon_channels(n_electron=n_electron,
+                                              z_obs=z,
+                                              positions=xy,
+                                              _instruction=_instruction,
+                                              config=config,
+                                              resource=resource) #if not len(_photon_timings)==0 else []
+
+        # Get photon timings including delays
+        _photon_timings = S2._get_photon_times(xy, n_photons_per_xy,
+                                               _photon_channels, config, phase, resource)
+        # repeat for n photons per electron # Should this be before adding delays?
+        _photon_timings += np.repeat(_electron_timings, n_photons_per_ele)
 
-        _photon_timings += Pulse.singlet_triplet_delays(
-            len(_photon_timings), config['singlet_fraction_gas'], config, phase)
+        # Remove photon with channel -1
+        mask = _photon_channels != -1
+        _photon_channels = _photon_channels[mask]
+        _photon_timings = _photon_timings[mask]
 
-        _photon_timings += np.random.normal(0, config['s2_time_spread'], len(_photon_timings)).astype(np.int64)
+        sorted_index = np.argsort(_photon_channels)
+        _photon_channels = _photon_channels[sorted_index]
+        _photon_timings = _photon_timings[sorted_index]
+        return _electron_timings, _photon_timings, _photon_channels
 
-        return _electron_timings, _photon_timings, _instruction
+    @staticmethod
+    def _get_photon_times(xy, n_photons_per_xy, channels, config, phase, resource):
+        """ Get photon times and add delays based on models
+        :param xy: 1d float array, xy positions of s2
+        :n_photons_per_xy:
+        :channels:
+        :param config: dict of the wfsim config
+        :param resource: instance of the resource class
+        """
+        # Luminescence Timings
+        if 'simple' in config['s2_luminescence_model']:
+            photon_timings = S2.luminescence_timings_simple(xy, n_photons_per_xy,
+                                                             config=config,
+                                                             resource=resource)
+        elif 'garfield' in config['s2_luminescence_model']:
+            confxy = True if 'confine distance' in config['s2_luminescence_model'] else False
```

Comment:
@JYangQi00 Hi Jianyang currently, I added it here. So while comparing the two, I simulate once setting `{'s2_luminescence_model':'garfield'}` and once with `{'s2_luminescence_model':'garfield confine distance'}` so both `confxy=True` and `confxy=False` are callable

Reply:
and [here](https://github.com/XENONnT/analysiscode/blob/master/SE_WFSim_shape/data-WFSim_comparison.ipynb) I uploaded a notebook, using this updates. I'm trying to compare the data vs WFSim with several configs